File: 3754.concatenate-non-zero-digits-and-multiply-by-sum-i/solution.kt
class Solution {
    fun sumAndMultiply(n: Int): Long {
        var sum = 0
        var x = 0L
        var p = 1
        var n = n

        while (n > 0) {
            val d = n % 10
            if (d != 0) {
                x += p * d
                p *= 10
                sum += d
            }
            n /= 10
        }

        return x * sum
    }
}

File: 3754.concatenate-non-zero-digits-and-multiply-by-sum-i/solution.go
func sumAndMultiply(n int) int64 {
    x, sum := 0, 0
    for f := 1; n > 0; n /= 10 {
        if d := n % 10; d > 0 {
            x += f * d
            sum += d
            f *= 10
        }
    }
    return int64(x) * int64(sum)
}